elasticsearch/connection/http_urllib3.py

```diff
@@ -62,33 +76,44 @@ def __init__(self, host='localhost', port=9200, http_auth=None,
                 self.headers[k.lower()] = headers[k]
 
         self.headers.setdefault('content-type', 'application/json')
-        ca_certs = CA_CERTS if ca_certs is None else ca_certs
         pool_class = urllib3.HTTPConnectionPool
         kw = {}
-        if use_ssl:
+
+        # if providing an SSL context, raise error if any other SSL related flag is used
+        if ssl_context and (use_ssl or verify_certs or ca_certs or ssl_version):
```

Comment:
If ssl_context is True, will not work, because use_ssl is true if https protocol is used.
This line will overwrite use_ssl to False or True depending on the protocol: https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/transport.py#L153, because of _normalize_hosts (https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/client/__init__.py#L170)
More details: https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/client/__init__.py#L19

Could you please remove use_ssl from the check, because when https is used use_ssl always true?

Reply:
Thanks for the feedback @zmathe

Reply:
@fxdgear Could you please also remove use_ssl from (use_ssl or verify_certs or ca_certs or ssl_version) tuple? Because if the protocol is https, use_ssl is True and can not set to False
Thanks again for making this available.